src/ui/wxWidgets/properties.h

```diff
@@ -150,6 +164,8 @@ class CProperties: public wxDialog
   wxString m_DbName;
   wxString m_DbDescription;
 ////@end CProperties member variables
+  StringX m_NewDbName;
+  StringX m_NewDbDescription;
```

Comment:
*m_DbName* and *m_DbDescription* are used to display data on the UI, whereas *m_NewDbName* and *m_NewDbDescription* reflect the data as it is currently set in the DB header, but also the data that will go finally into the DB, if it was changed.
'N/A' should be shown at the UI as long as name and description are not set or when they are empty strings.

The user might even set explicitly the name and description to 'N/A' so that 'N/A' needs to be shown at the UI, but also the string 'N/A' needs to be used for the DB property name or description. Another scenario is that the user left an empty string in the edit dialog. In that case 'N/A' will be shown on the UI but the header property will be set to an empty string. So, one purpose is to handle the case of 'N/A' and an empty string. Separat variables make it easy to distinguish between what to display on the UI and what to pass to the DB.

Also if the validation of a given string for DB name or DB description fails, *m_NewDbName* and *m_NewDbDescription* won't get updated, so that invalid data will not make its way into the DB header. *m_NewDbName* and *m_NewDbDescription* represent header conform data and are compared against the current header data to figure out whether anything has changed. - That's at least the idea, but I'm open for improvements.

Reply:
The reason for this additional requirement is that from my viewpoint as long as DB name and DB description are just set to an empty string or in other words have never been set this means that these information are undefined or not available. I would expect that this case is indicated always with "N/A" on the UI, which is only done if the properties dialog is opened. If the properties dialog is already open and the user explicitly sets from this dialog DB name or DB description to an empty string nothing would appear on the UI, because of the empty string. If the properties dialog is opened then once again from the applications main menu, the UI displays suddenly "N/A", which is in my opinion inconsistent. But because it's also the behaviour of the Windows version I will change it according to your proposal.

Reply:
To not have additional variables it becomes quite difficult to check properly for any property changes and will result in the following two, in my opinion undesired, scenarios.

#### Case A)
1. The user opens a DB with both properties, DB name and DB description, not set so far to any string.
1. Now, the user opens the properties dialog, just to have a look on the DB properties.
1. The members m_DbName and m_DbDescription get both initialized with the string "N/A" to display this one on the UI, indicating that the properties aren't set.
1. The user closes the properties dialog without having applied any changes.
1. Now, the user is faced with a modified DB, because m_DbName/m_DbDescription (="N/A") differ from the content (still empty strings) of the DB header, due to the member initialization (see 3)

#### Case B)
1. The user opens a DB with DB name and DB description set to a string.
1. The user opens the properties dialog to edit both properties. She/He deletes them which results in getting back an empty string from the properties edit dialogs.
1. On the UI no string appears even not "N/A".
1. If the properties dialog is reopened then "N/A" is shown as replacement for the empty strings.

This is from my viewpoint an inconsistent and user confusing behaviour.

I have no idea how to implement a check for changes concerning the DB name and DB description without having some additional variables. Without those the DB header gets modified even if the user hasn't explicitly modified anything in case DB name/description are empty strings. Additionally, if the user wants to reset DB name/description back to an empty string then after removing those strings and closing the edit dialog no string is shown on the properties dialog.

Suggestions are welcome.

Reply:
I see two options to resolve the issues above:

1. Drop the requirement that the default UI text should be "N/A" and let it be an empty string just like stored in the DB.
2. I'm not very familiar with wxWidgets, but in many web frameworks the stored data values and the way they're rendered can be different, so e.g. it's possible to define placeholder values at the UI level. I don't know if this is possible here, but conceptually we have two different places the value is displayed:
  * the DB properties view dialog - this is a read-only representation that can in theory display values distinct from what's stored
  * the edit name/description dialogs - here we need to show the exact value stored, and whatever the user enters is saved (as long as it's valid)

That said I don't feel strongly about these two extra fields and don't want to hold up merging this change.

Reply:
Note that storing an empty string is more "friendly" for multilingual support. "N/A" is English, and perhaps French...